scripts/test_env.sh

```diff
@@ -97,6 +97,8 @@ function run_peer () {
     STORAGE="$PEER/storage"
     mkdir -p "$STORAGE"
     KURA_BLOCK_STORE_PATH="$STORAGE"
+	LOG_FILE_PATH="$PEER/log.json"
+	MAX_LOG_LEVEL="TRACE"
```

Comment:
Speaking about the `test_env` script, it seems to fail for me when I run it with either the current `iroha2-dev` hash or likewise on this branch, complaining about failing to build the proxy without the `debug_force_soft_fork` being set -- so it was introduced by. Docker compose seems is still building fine. So it feels like despite what was mentioned in the soft-fork PR [discussion](https://github.com/hyperledger/iroha/pull/3064#discussion_r1073378147), it's still being exposed technically, as it's become part of the `ConfigurationProxy` via the derive macro. Should we create an issue for this?

Reply:
Aha! 

So here's my take on why this is. I'll try to fix it in this PR. 

Whe @SamHSmith was implementing the fix for #1170, we found that it wasn't possible to hide the configuration variable behind a feature flag, because of how `cargo` resolves optional features for multiple configurations. We ended up saying that the configuration parameter should only be present if `cfg(debug_assertions)` is also enabled. So a debug build (which we use for the testing) also retains that parameter as a requirement. 

Stop-gap solution is to add an env-var to silence the warning. 

Better solution is to make the variable optional and if it's not provided, set it to false. 

I'd also add a 
```rust 
warn!(%force_soft_forks=config.debug_force_soft_fork, "Running Iroha Debug build")
```
Just to be safe.